File: problems 1 to 10/p1-reverse-an-array/ans.java
/*
-> Given an array (or string), the task is to reverse the array/string.
// Examples : 


// Input  : arr[] = {1, 2, 3}
// Output : arr[] = {3, 2, 1}

// Input :  arr[] = {4, 5, 1, 2}
// Output : arr[] = {2, 1, 5, 4}
*/ 

public class ans{
    static void reverse(int arr[],int start,int end){
        int temp=0;
        while(start<end-1){
            temp=arr[start];
            arr[start]=arr[end];
            arr[end]=temp;
            start++;end--;
        }

    }
    static void printarr(int arr[],int size){
        for (int i = 0; i < size; i++)
        {   
            if(i==0) System.out.print("[");
            System.out.print(arr[i]);
            if(i!=size-1) System.out.print(",");
            if(i==size-1) System.out.print("]\n");
        
        }
    }

    public static void main(String[] args){
        int arr[]={1,2,3,4,5};
        
        System.out.println("finally");
        printarr(arr,5);
        reverse(arr, 0, 4);
        printarr(arr,5);
    }
}

File: problems 1 to 10/p1-reverse-an-array/ans.c
#include<stdio.h>
/*
-> Given an array (or string), the task is to reverse the array/string.
// Examples : 


// Input  : arr[] = {1, 2, 3}
// Output : arr[] = {3, 2, 1}

// Input :  arr[] = {4, 5, 1, 2}
// Output : arr[] = {2, 1, 5, 4}
*/ 
void printarr(int arr[],int size){
    for (int i = 0; i < size; i++)
    {   
        (i==0) ? printf("["):printf(",");
        printf(" %d",arr[i] );
        (i==size-1) ? printf("]\n"):printf("");
    }
    
}
void swap(int *a, int* b){
    int temp=*a;
    *a=*b;
    *b=temp;

}

void reverse(int arr[],int start,int end){
    int temp=0;
    while (start<end){
        swap(&arr[start],&arr[end]);
        start++;
        end--;
    }
}
int main(){
    int arr[5]={1,2,3,4,5};
    printarr(arr,5);
    reverse(arr,0,4);
    printarr(arr,5);
    return 0;
}